src/nix.rs

```diff
@@ -250,53 +238,49 @@ impl ToNickel for rnix::SyntaxNode {
             )
             .into(),
-            ParsedType::IfElse(n) => if_then_else(
+            Expr::IfElse(n) => if_then_else(
                 n.condition().unwrap().translate(state),
                 n.body().unwrap().translate(state),
                 n.else_body().unwrap().translate(state),
             ),
-            ParsedType::BinOp(n) => n.translate(state),
-            ParsedType::UnaryOp(n) => n.translate(state),
-            ParsedType::Select(n) => match ParsedType::try_from(n.index().unwrap()).unwrap() {
-                ParsedType::Ident(id) => Term::Op1(
-                    UnaryOp::StaticAccess(id.as_str().into()),
-                    n.set().unwrap().translate(state),
-                )
-                .into(),
-                _ => Term::Op2(
-                    BinaryOp::DynAccess(),
-                    n.index().unwrap().translate(state),
-                    n.set().unwrap().translate(state),
-                )
-                .into(),
-            },
-            ParsedType::OrDefault(n) => unimplemented!(),
-
-            // TODO: what are these?
-            ParsedType::PatBind(_) | ParsedType::PathWithInterpol(_) => {
-                unimplemented!()
+            Expr::BinOp(n) => n.translate(state),
+            Expr::UnaryOp(n) => n.translate(state),
+            Expr::Select(n) => {
+                n.attrpath()
+                    .unwrap()
+                    .attrs()
+                    .fold(n.expr().unwrap().translate(state), |acc, i| {
+                        match i {
+                            rnix::ast::Attr::Ident(id) => {
+                                Term::Op1(UnaryOp::StaticAccess(id.to_string().into()), acc)
+                            }
+                            rnix::ast::Attr::Dynamic(d) => Term::Op2(
+                                BinaryOp::DynAccess(),
+                                d.expr().unwrap().translate(state),
+                                acc,
+                            ),
+                            rnix::ast::Attr::Str(s) => {
+                                Term::Op2(BinaryOp::DynAccess(), s.translate(state), acc)
+                            }
+                        }
+                        .into()
+                    })
             }
-
-            // The following nodes aren't `RichTerm`s but sub-parts of other nodes higher in the syntax tree.
-            // They can't be parsed independently of a term and should never occur at this level of the translation.
-            ParsedType::Pattern(_)
-            | ParsedType::PatEntry(_)
-            | ParsedType::Inherit(..)
-            | ParsedType::InheritFrom(..)
-            | ParsedType::Key(..)
-            | ParsedType::KeyValue(..) => unreachable!(),
+            Expr::Path(_) | Expr::HasAttr(_) => unimplemented!(), // TODO: What is the diff with
+                                                                  // Select / operator HasField?
```

Comment:
I didn't understand this comment.

Reply:
Sorry:

- For the `HasAttr` I discovered later that it's not an operator anymore in `rnix 0.11`. It was an operator before but now it's a specific language construct.
- But for `Expr::Path` I don't understand what it is for. we already have `Select` which is the `.` operator. So what is a `Expr::Path`? 

`rnix` is a very good crate but sadly without any documentation...

Reply:
@francois-caddet I suppose it's a file path, like `/foo/bar/baz`, not something related to attribute paths?

Reply:
Yes but so what is `Literal::Uri(_)`? 😢 
I will merge as it is and see what it is adding tests in the main branch `nix/nickel`